docs/frontends/queues.md

```diff
@@ -81,26 +79,64 @@ Using a circular buffer usually entails incrementing indices modulo the buffer s
+client can divide the incoming traffic into `n` flows.
+For example, if `n = 3` and the client passes boundaries `[133, 266, 400]`,
+packets will be divided into three flows according to the intervals: `[0, 133]`, `[134, 266]`, `[267, 400]`.
+
+- At `push`, we check the `boundaries` list to determine which flow to push to.
+Take the boundaries example given earlier, `[133, 266, 400]`.
+If we push the value `89`, it will, under the hood, be pushed into subqueue 0 becuase `0 <= 89 <= 133`,
+and `305` will be pushed into subqueue 2 since `266 <= 305 <= 400`.
+- The program maintains a `hot` pointer that starts off at 0, meaning the next subqueue to pop from is queue 0.
+At `pop` we first try to pop from `hot`. If this succeeds, great. If it fails,
+we increment `hot` and therefore continue to check all other flows
+in round robin fashion.
+
+The source code is available in [`gen_strict_or_rr.py`][gen_strict_or_rr.py], which takes as arguments `n`, `boundaries`, and handles to the subqueues it must administer. It also takes a boolean parameter `round_robin`, which, if `true`, results in the generation of a round-robin queue.
+
+
+### Strict Queues
+
+Strict queues support `n` flows as well, but instead, flows have a strict order of priority and this which determines popping
+order. That is, the second-highest priority subqueue will only be allowed to pop if the highest priority subqueue is empty.
+If the higher-priority flow get pushed to in the interim, the next call to `pop` will again try to pop from the highest priority flow.
+
+Like round-robin queues, it takes in a list `boundaries` that must be of length `n`, which divide the incoming traffic into `n` flows.
+For example, if `n = 3` and the client passes boundaries `[133, 266, 400]`,
+packets will be divided into three flows according to the intervals: `[0, 133]`, `[134, 266]`, `[267, 400]`.
+
+It takes a list `order` that must be of length `n`, which specifies the order
+of priority of the flows. For example, if `n = 3` and the client passes order
+`[1, 2, 0]`, then flow 1 (packets in range `[134, 266]`) has first priority, flow 2
+(packets in range `[267, 400]`) has second priority, and flow 0 (packets in range
+`[0, 133]`) has last priority.
+
+- At push, we check the `boundaries` list to determine which flow to push to.
+Take the boundaries example given earlier, `[133, 266, 400]`.
+If we push the value `89`, it will, under the hood, be pushed into subqueue 0 becuase `0 <= 89 <= 133`,
+and `305` will be pushed into subqueue 2 since `266 <= 305 <= 400`.
+- Pop first tries to pop from `order[0]`. If this succeeds, great. If it fails, it tries `order[1]`, and so on.
+
+The source code is available in [`gen_strict_or_rr.py`][gen_strict_or_rr.py], which takes as arguments `n`, `boundaries`, `order`, and handles to the subqueues it must administer. It also takes a boolean parameter `round_robin`, which, if `false`, results in the generation of a strict queue.
 
 ## PIFO Tree
```

Comment:
Only request: in the text of this, please also give a link to the more complicate PIFO tree you recently did up!